### PR TITLE
TE 3.2: add a flag to enabl/diable adding a drop route

### PR DESCRIPTION
**Pull request by mojiiba**
This pull simply adds a flag to enable/disable adding drop route. Also, please note that the semantics of drop route is a vendor dependent and the expected behavior may need to be captured by static route tests.

### CODE REVIEW

**Review by xw-g — CHANGES_REQUESTED** [review]

File: feature/gribi/ate_tests/weighted_balancing_test/setup_test.go [hunk 1]
```diff
@@ -174,16 +176,18 @@ func configureDUT(t testing.TB, dut *ondatra.DUTDevice) {
 	// down, the device does not attempt to route packets through the
 	// default gateway 0.0.0.0/0.  Packets destined to the more specific
 	// next hop CIDRs will be routed.
-	static := &telemetry.NetworkInstance_Protocol_Static{
-		Prefix: ygot.String(discardCIDR),
+	if *addNullRoute {
+		static := &telemetry.NetworkInstance_Protocol_Static{
+			Prefix: ygot.String(discardCIDR),
+		}
+		static.GetOrCreateNextHop("AUTO_drop_2").
```
[the remark this answers is not in the document]
Comment (mojiiba) [reply feature/gribi/ate_tests/weighted_balancing_test/setup_test.go:183]: sure

File: feature/gribi/ate_tests/weighted_balancing_test/setup_test.go [hunk 2]
```diff
@@ -46,6 +46,8 @@ var (
 		"Amount of time to pause before sending traffic.")
 	trafficDuration = flag.Duration("traffic_duration", 5*time.Second,
 		"Duration for sending traffic.")
+
+	addNullRoute = flag.Bool("add_drop_route", true, "add a drop route so that when the nexthop interface goes down, the device does not attempt to route packets through the default gateway 0.0.0.0/0.")
```
[the remark this answers is not in the document]
Comment (mojiiba) [reply feature/gribi/ate_tests/weighted_balancing_test/setup_test.go:50]: the way the discard route is added in this test is not vendor neutral and this does not work as discard route in our devices.  @aaronmillisor can give you more details on this.

[the remark this answers is not in the document]
Comment (xw-g) [reply feature/gribi/ate_tests/weighted_balancing_test/setup_test.go:50]: Yes, deviation flag MUST be a global variable in the deviation.go

[the remark this answers is not in the document]
Comment (mojiiba) [reply feature/gribi/ate_tests/weighted_balancing_test/setup_test.go]: thanks, addressed your comment

File: feature/gribi/ate_tests/weighted_balancing_test/setup_test.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (mojiiba) [reply feature/gribi/ate_tests/weighted_balancing_test/setup_test.go]: done

[the remark this answers is not in the document]
Comment (mojiiba) [reply feature/gribi/ate_tests/weighted_balancing_test/setup_test.go]: @xw-g please let me if you agree with the changes?

**Comment by xw-g** [comment]
@vosipchu is investigating some possible augmentation in the infra that we might not need this deviation.

**Comment by mojiiba** [comment]
closing this pr since the default route will be removed.

**Pull request closed without merging**